### PR TITLE
boards: stk3700: add support (v2)

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description

This PR add supports for the Silicon Labs [STK3700](https://github.com/basilfx/EFM2RIOT/blob/master/dist/doc/STK3700.md). In this new version, common files have been moved to `boards/common/silabs`.

The files are very similar to the SLTB001a already supported.

### Issues/PRs references

Made possible by #8139, successor of #8297, with recommendations of #8295.

### CODE REVIEW

**Comment by kYc0o** [comment]
BTW, I tested and works very good so far, besides the xtimer config which may need some tuning.

**Comment by kYc0o** [comment]
Well, I think this one needs rebase :)

**Comment by basilfx** [comment]
Rebased and adapted to boards/common/silabs.

File: boards/stk3700/board.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2015-2018 Freie Universität Berlin
```
Comment (kYc0o) [inline boards/stk3700/board.c]: I'd say the copyright belongs to you too?

**Comment by kYc0o** [comment]
Besides the "missing" (I'd say optional) copyrights I found a non working RTC. If you want to fix it is OK, otherwise you can remove it from the list of the provided features and work on it later. This way at least this PR is merged.

File: boards/stk3700/board.c (hunk 1 above)
Comment (kYc0o) [reply boards/stk3700/board.c]: Ok no problem then.

**Comment by basilfx** [comment]
Created #8572 to address RTC and ADC issue. Will wait for that one.

**Comment by kYc0o** [comment]
Small rebase needed again 😅

**Comment by basilfx** [comment]
Rebased.

**Pull request merged**